gubernator/README.md

```diff
@@ -29,3 +29,71 @@ For deployment:
         }
     }
 ```
+
+# GCS Layout API
+
+In order to correctly interpret jobs results, in GCS, Gubernator expects that
+any one job directory is laid out in a specific manner, and that job directories
+are laid out in a specific manner relative to each other.
+
+## Job Artifact GCS Layout
+
+For a single build of a job, Gubernator expects the following layout in GCS:
+
+```
+.
+├── artifacts         # all artifacts must be placed under this directory
+│   └── junit_00.xml  # jUnit XML reports from the build
+├── build-log.txt     # std{out,err} from the build
+├── finished.json     # metadata uploaded once the build finishes
+└── started.json      # metadata uploaded once the build starts
+```
+
+The following fields in `started.json` are honored:
+
+```json
+{
+    "timestamp": "seconds after UNIX epoch that the build started",
+    "pull": "$PULL_REFS from the run",
+    "repos": {
+        "org/repo": "git version of the repo used in the test",
+    },
+}
+```
+
+The following fields in `finished.json` are honored:
+
+```json
+{
+    "timestamp": "seconds after UNIX epoch that the build finished",
+    "result": "human-readable string detailing the result of the build",
+    "passed": "boolean, whether the build succeeded",
+}
```

Comment:
`"metadata": "dictionary of additional key-value pairs that will be displayed to the user"`